rskj-core/src/test/java/org/ethereum/rpc/Web3ImplTest.java

```diff
@@ -1230,6 +1256,44 @@ public void eth_sendTransaction()
         Assert.assertTrue("Method is not creating the expected transaction", expectedHash.compareTo(txHash) == 0);
     }
 
+    @Test
+    public void eth_sendTooMuchGasTransaction()
+    {
+        BigInteger nonce = BigInteger.ONE;
+        Web3Impl web3 = createWeb3();
+
+        // **** Initializes data ******************
+        String addr1 = web3.personal_newAccountWithSeed("sampleSeed1");
+        String addr2 = web3.personal_newAccountWithSeed("sampleSeed2");
+
+        BigInteger value = BigInteger.valueOf(7);
+        BigInteger gasPrice = BigInteger.valueOf(8);
+        BigInteger gasLimit = new BigDecimal(Math.pow(2, 63)).toBigInteger();
+
+        String data = "0xff";
+
+        // ***** Executes the transaction *******************
+        Web3.CallArguments args = new Web3.CallArguments();
+        args.from = addr1;
+        args.to = addr2;
+        args.data = data;
+        args.gas = TypeConverter.toJsonHex(gasLimit);
+        args.gasPrice = TypeConverter.toJsonHex(gasPrice);
+        args.value = value.toString();
+        args.nonce = nonce.toString();
+
+        String txHash = null;
+        try {
+            txHash = web3.eth_sendTransaction(args);
+        }
+        catch (JsonRpcInvalidParamException e) {
+            Assert.assertEquals(e.getMessage(), "Gas limit exceeds Transaction max value tolerated");
+        } catch (Exception e) {
+            Assert.assertFalse(e instanceof JsonRpcInvalidParamException);
+        }
+        Assert.assertNull(txHash);
```

Comment:
could we use an Assert.fail? And then, in the first catch, add a return command, then the second catch could be removed

Reply:
I'll improve this so it's a bit cleaner